lib/Sema/SemaExpr.cpp

```diff
@@ -7401,7 +7414,12 @@ checkPointerTypesForAssignment(Sema &S, QualType LHSType, QualType RHSType) {
   // between checked and unchecked pointers or between different kinds of checked
   // pointers. Only implicit conversions between unchecked pointers or from
   // unchecked to checked pointers are allowed.
-  if (lhkind != rhkind && rhkind != CheckedPointerKind::Unchecked)
+  // Also, implicit conversion between checked pointers is added
+  // to handle properly address-of (&) operator in checked block
+  // Except for implicit conversion from checked to unchecked pointers,
+  // all other implicit conversions are allowed
+  if (lhkind != rhkind && rhkind != CheckedPointerKind::Unchecked &&
```

Comment:
Can this just be simplified to rhkind != CheckedPointerKind::Unchecked &&
lhKind == CheckedPointerKind::Unchecked?   I think those two conditions imply that lhkind != rhkind, so that clause can be deleted.